examples/app/flight_delays/main.py

```diff
@@ -0,0 +1,144 @@
+import os
+import datetime
+import numpy as np
+import pandas as pd
+
+from bokeh.models import HBox, VBox
+from bokeh.io import curdoc
+from bokeh.plotting import Figure
+from bokeh.models.widgets import Select, Slider
+from bokeh.models import ColumnDataSource, HoverTool, Paragraph
+from bokeh.palettes import Reds5
+
+import itertools
+
+from bokeh.sampledata.us_states import data as states
+
+def load_data():
+
+    # load states
+    states.pop("HI", None)
+    states.pop("AK", None)
+    state_xs = [states[code]["lons"] for code in states]
+    state_ys = [states[code]["lats"] for code in states]
+    x_chain = list(itertools.chain(*state_xs))
+
+    max_long = max(x_chain)
+    min_long = min(x_chain)
+
+    # load airlines delays
+    airline_delays_df = pd.read_csv(os.path.join(os.path.dirname(__file__), 'data/airline_delay_causes.csv'))
+    airline_delays_df['period'] = airline_delays_df.apply(lambda row: datetime.date(month=row[' month'], year=row['year'], day=1), axis=1)
+    airline_delays_df = airline_delays_df[['period', 'airport', 'arr_flights', 'arr_del15', 'carrier_name']]
+
+    # load airport locations
+    adf = pd.read_csv(os.path.join(os.path.dirname(__file__), 'data/airports.csv'))
+    adf = adf[adf['country'] == 'United States'][['iata/faa', 'latitude', 'longitude', 'name']]
+    adf = adf[(adf.longitude < max_long) & (adf.longitude > min_long)]
+    adf.set_index('iata/faa', inplace=True)
+
+    return state_xs, state_ys, airline_delays_df, adf
+
+def on_year_change(attr, old, new):
+    global year
+    year = new
+    update_map()
```

Comment:
globals not needed. Just call `update_map` and have `update_map` read the values off the widget objects directly.